src/wallet/mod.rs

```diff
@@ -1867,8 +1870,9 @@ where
 }
 
 /// Return a fake wallet that appears to be funded for testing.
-pub fn get_funded_wallet(
+pub fn get_funded_wallet_with_coinbase(
     descriptor: &str,
+    coinbase: bool,
```

Comment:
Why not just always call coinbase?? There is no way for us to ever specify any inputs in `get_funded_wallet()` anyway.. So doesn't hurt to always have the coinbase input than empty inputs, which is not a valid transaction.. 

And having the funding transaction, coinbase, even if its fake, makes semantic sense..

Reply:
> And having the funding transaction, coinbase, even if its fake, makes semantic sense..

I agree with that, 100%. However, it creates some problems when height is specified in a test, as coinbase inputs need to mature before being used. A couple tests were failing if I used coinbase in all tests.

Reply:
Right.. And this gets fixed by just adding a 100 to the height values of those tests..

Reply:
Okay. I was hoping I could get by without changing the test... I will fix them.

Reply:
Yeah those are out of context a bit.. But small changes only in two tests is not that critical IMO.. Thanks for the quick updates.. I will go over the fully once more..

Reply:
Sorry if I'm late to the conversation, but can't we just make sure that in `populate_test_db` we create a sane transaction with inputs in all cases?

```diff
diff --git a/src/database/memory.rs b/src/database/memory.rs
index 7d806eb..ad1eda9 100644
--- a/src/database/memory.rs
+++ b/src/database/memory.rs
@@ -490,11 +490,10 @@ macro_rules! populate_test_db {
         let mut db = $db;
         let tx_meta = $tx_meta;
         let current_height: Option<u32> = $current_height;
-        let input = if $is_coinbase {
-            vec![$crate::bitcoin::TxIn::default()]
-        } else {
-            vec![]
-        };
+        let mut input = vec![$crate::bitcoin::TxIn::default()];
+        if !$is_coinbase {
+            input[0].previous_output.vout = 0;
+        }
         let tx = $crate::bitcoin::Transaction {
             version: 1,
             lock_time: 0,
```

So that we don't have to modify tests...

What do you think?

Reply:
One thing that comes to mind while setting the input in `populate_test_db` is, what if we need to insert a fake transaction in the db, where the input matters? Even if we set them in `tx_meta` of `testutils`, it will get ignored in this function.. And we will always have a specific input in any transaction added via populate_test_db.. 

What I think can be more useful (I partially tried this in #624) make sure populate test db creates the transaction as per the `tx_meta` passed in.. If it doesn't have inputs (which is the case for `get_funded_wallet`) make it a coinbase..

Reply:
> make sure populate test db creates the transaction as per the tx_meta passed in..

Right, that's a good idea!

>  If it doesn't have inputs (which is the case for get_funded_wallet) make it a coinbase..

What about the opposite: if it doesn't have the inputs, we use a dummy default (like all zeros for the txid, and 0 for the vout, so it's not a coinbase) - I think this would be cleaner, as:
- We rarely need coinbase inputs for tests
- Coinbase inputs are easy to specify (just use `Outpoint::null()`)
- Coinbase inputs might bring some troubles in tests (you need to have 100 more confirmations etc)

Reply:
That makes total sense to me.. I don't think we will need this input in any situation, and for coinbase situations we are already specifying them.. ACK on that will be more cleaner..